### ISSUE TITLE
Add the ability to associate an object to the match.

### ISSUE THREAD

**Issue by Mistobaan** [schollz/closestmatch#2]
it would be useful to associate an `interface{}` to the word, or return an index of the match.

**Comment by Yugloocamai** [comment]
Yes, this would be GREAT!  I expected this actually, and just ran into a wall.

**Comment by Yugloocamai** [comment]
@Mistobaan I forked and added interface associations, might be helpful to you.  https://github.com/Yugloocamai/closestmatch

**Comment by schollz** [comment]
@Yugloocamai I think I'd rather not modify this version, as it will break some things I've made that depend on it. Would you like to make a PR in this project's README to guide people to your project?
